### PR TITLE
Move TypeTuple alias to std.typetuple.

### PR DESCRIPTION
**Pull request by deadalnix**
There is no reason to expose this more than it needs to. This ensure the role of backward compatibility already.

### CODE REVIEW

**Comment by schveiguy** [comment]
Seems it's not that simple -- there's a lot of code in std.meta that still uses the `TypeTuple` name.

**Comment by deadalnix** [comment]
@schveiguy It seems that this code also import std.typetuple. But This fails because of an ICE, interesting :)

**Comment by schveiguy** [comment]
> this code also import std.typetuple

I think that's a relic from moving code from a different module. That import should be removed.

**Comment by deadalnix** [comment]
OK I just changed std.meta to use Arguements and added unitest block for TypeTuple.

**Comment by schveiguy** [comment]
I see many instances of "typetuple" in the comments. I think you are right that "Arguments" is a poor name, because just substituting Arguments doesn't read very well.

One obvious change that should be made, the macro on line 28 to link to the source (links to std/typetuple.d)

**Comment by deadalnix** [comment]
Can we move on with that one, and I can prepare another diff changing the typetuple/argument name in the code/comments. I think these are 2 different problem (not exposing the typetuple symbol when it is not necessary for backward compatibility, and having a good name for it).

**Comment by schveiguy** [comment]
I think maybe you misunderstand. The documentation is referring to a `typetuple` which doesn't exist anymore. It's confusing documentation. Changing the name of `Arguments` is definitely separate, but I think it's wrong to have all these references to something that no longer exists in the docs. Change them to `Arguments`. It may look awkward, but at least it's referring to a real symbol.

**Comment by deadalnix** [comment]
I could find only one reference to the actual TypeTuple symbol, which is gone. I just fixed it.

**Comment by schveiguy** [comment]
Hm... I still see a lot of cases of typetuple. For instance, https://github.com/D-Programming-Language/phobos/pull/3440/files#diff-33b50b4967214bfd07ca2ebb7bbc023cR202

And of course, the reference to std.typetuple source on line 28 is still there.

**Comment by deadalnix** [comment]
This refers to typetuple the 'concept' rather than TypeTuple the 'symbol'. IMO that belong to the renaming phase.

File: std/meta.d [hunk 1]
```diff
@@ -41,7 +41,6 @@ template Arguments(TList...)
 ///
 unittest
 {
-    import std.typetuple;
```
Comment (CyberShadow) [inline std/meta.d]: Maybe this should say `import std.meta` instead, to be a self-contained example?

**Comment by schveiguy** [comment]
Where is typetuple the 'concept' defined?

**Comment by deadalnix** [comment]
@schveiguy nowhere, this is why everybody is confused about it.

File: std/meta.d (hunk 1 above)
Comment (schveiguy) [inline std/meta.d]: I disagree. We don't do that anywhere else (even in this file)

**Comment by CyberShadow** [comment]
We have a page on dlang.org:

http://dlang.org/tuple.html

File: std/meta.d (hunk 1 above)
Comment (CyberShadow) [inline std/meta.d]: OK.

**Comment by schveiguy** [comment]
OK, well at least there is something we can point people at. I don't think it's necessarily a type tuple though.

File: std/meta.d (hunk 1 above)
Comment (deadalnix) [inline std/meta.d]: Should it be removed then ?

**Comment by CyberShadow** [comment]
Is this good to merge (to pave the way for the `Arguments` rename)?

**Comment by schveiguy** [comment]
Yeah, I can fix the ddoc reference to the wrong file in a separate pull.

**Comment by schveiguy** [comment]
Auto-merge toggled on

**Pull request merged**